Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp

```diff
@@ -693,3 +679,31 @@ void CVehicleRPCs::SetVehicleNitroActivated(CClientEntity* pSourceEntity, NetBit
         vehicle->SetNitroLevel(vehicle->GetNitroLevel() + 1.0001f);
 }
 
+void CVehicleRPCs::SetVehicleRotorSpeed(CClientEntity* pSource, NetBitStreamInterface& bitStream)
+{
+    CClientVehicle* pVehicle = m_pVehicleManager->Get(pSource->GetID());
+    if (pVehicle)
+    {
+        unsigned char ucRotorSpeed;
+        if (bitStream.Read(ucRotorSpeed))
+        {
+            // Convert the given rotor speed from between 0-100 to 0-0.22
+            float rotorSpeed = (static_cast<float>(ucRotorSpeed) / 100.0f * 0.22f);
+            pVehicle->SetRotorSpeed(rotorSpeed);
+        }
+    }
+}
+
+void CVehicleRPCs::SetVehicleRotorState(CClientEntity* pSource, NetBitStreamInterface& bitStream)
+{
+    CClientVehicle* pVehicle = m_pVehicleManager->Get(pSource->GetID());
+    if (pVehicle)
+    {
+        bool rotorState;
+        bool stopRotor;
+        if (bitStream.ReadBit(rotorState) && bitStream.ReadBit(stopRotor))
```

Comment:
ReadBit returns true/false, so if you call setVehicleRotorState with state = true and stopRotor = false, this condition will never be met.